harmony/harmony.py

```diff
@@ -361,6 +362,49 @@ def _params(self, request: Request) -> dict:
 
         return params
 
+    def _headers(self) -> dict:
+        """
+        Create (if needed) and return a dictionary of headers.
+        Code partially adaped from:
+            https://github.com/requests/toolbelt/blob/master/requests_toolbelt/utils/user_agent.py
+        """
+        if 'headers' not in self.__dict__:
+            user_agent_content = set([])
+
+            # Get harmony package info
+            try:
```

Comment:
I think you can just use one `try` block for all three since the exception handling is identical.

Reply:
Yes, will do.

Reply:
Done.

Reply:
Hmm, had a second thought, maybe we do want to keep separate `try` blocks, because I think it's Okay to miss some of the user-agent header items. If we merge them into one block, we may miss some header items to be generated after those throwing exceptions.

Reply:
Reverted back to "separate try blocks".

Reply:
I don't understand why this is in a try block in the first place or why the `from` statement isn't at the top of the file, since the import should be completely under our control.  In what case could an exception be thrown?

Reply:
Just moved this import to the top of the file.